cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs

```diff
@@ -1281,16 +1228,13 @@ runQueryStakePoolsCmd
     join $
       lift
         ( executeLocalStateQueryExpr localNodeConnInfo target $ runExceptT @QueryCmdError $ do
-            AnyCardanoEra era <- lift queryCurrentEra & onLeft (left . QueryCmdUnsupportedNtcVersion)
+            AnyCardanoEra era <- easyRunQueryCurrentEra
 
             sbe <-
               requireShelleyBasedEra era
                 & onNothing (left QueryCmdByronEra)
 
-            poolIds <-
-              lift (queryStakePools sbe)
-                & onLeft (left . QueryCmdUnsupportedNtcVersion)
-                & onLeft (left . QueryCmdEraMismatch)
```

Comment:
This instance doesn't seem equivalent because it doesn't seem to handle the same errors. Before the function extraction it handled `QueryCmdEraMismatch`, and `easyRunQuery` doesn't (it handles `QueryCmdLocalStateQueryError . EraMismatchError`).

Reply:
@palas> the change was correct because actually `QueryCmdLocalStateQueryError . EraMismatchError` was equivalent to `QueryCmdEraMismatch`. But your comment made me realize that the `QueryCmdEraMismatch` approach was simpler, so I added a commit deleting `QueryCmdLocalStateQueryError` altogether, because it had only one use (that we can replace with `QueryCmdEraMismatch`). Can you have a new look?